extensions/ql-vscode/src/run-queries.ts

```diff
@@ -199,6 +209,30 @@ export class QueryEvaluationInfo {
     };
   }
 
+  /**
+   * Prints the end of the structured log summary file (starting from most expensive predicates)
+   * to the query server Output Window.
+   * @param qs The current QueryServerClient instance for this query.
+   */
+  printEvalLogSummary(qs: qsClient.QueryServerClient) {
+    if (this.hasEvalLogSummary()) {
+      fs.readFile(this.evalLogSummaryPath, (err, buffer) => {
```

Comment:
Note that `fs.readFile` will read the whole file into memory which for very long evaluation runs could potentially cause problems, especially for users with not much RAM on their machine. Anecdotally, I've seen these logs get to a few hundred megabytes before, and we're storing them twice here at a point because they'll be in both `buffer` and `summaryText` after we've done `const summaryText = buffer.toString()`. Reading using [streams](https://nodejs.dev/learn/nodejs-streams) would be a better option to keep our memory consumption down, especially since we're throwing most of the file out anyway.